src/Microsoft.Extensions.CodeGeneration.Core/DefaultCodeGeneratorAssemblyProvider.cs

```diff
@@ -54,8 +51,12 @@ public IEnumerable<Assembly> CandidateAssemblies
                     .SelectMany(_libraryManager.GetReferencingLibraries)
                     .Distinct()
                     .Where(IsCandidateLibrary);
-                
+#if NET451
+                // For Net 451 the exe is being invoked from the build folder so all dependencies are next to the exe.
+                return list.Select(lib => _assemblyLoadContext.LoadFromName(new AssemblyName(lib.Identity.Name)));
```

Comment:
For scaffolding would it really be cumbersom to dispatch for both? It'd cleanup your code paths significantly in this PR.